app/services/LocationService.js

```diff
@@ -383,70 +423,93 @@ export default class LocationServices {
-        //           onPress: () => console.log('No Pressed'),
-        //           style: 'cancel',
-        //         },
-        //       ],
-        //     ),
-        //   1000,
-        // );
+        //we need to set delay or otherwise alert may not be shown
+        setTimeout(
+          () =>
+            Alert.alert(
+              languages.t('label.require_location_services_title'),
+              languages.t('label.require_location_services_message'),
+              [
+                {
+                  text: languages.t('label.yes'),
+                  onPress: () => {
+                    if (isPlatformAndroid()) {
+                      // showLocationSettings() only works for android
+                      BackgroundGeolocation.showLocationSettings();
+                    } else {
+                      // eslint-disable-next-line
+                      Linking.openURL('App-Prefs:Privacy'); // Deeplinking method for iOS
+                    }
+                  },
+                },
+                {
+                  text: languages.t('label.no'),
+                  onPress: () => {
+                    console.log('No Pressed'), LocationServices.stop();
+                  },
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
       } else if (!status.authorization) {
         // we need to set delay or otherwise alert may not be shown
-        // setTimeout(
-        //   () =>
-        //     Alert.alert(
-        //       languages.t('label.require_location_information_title'),
-        //       languages.t('label.require_location_information_message'),
-        //       [
-        //         {
-        //           text: languages.t('label.yes'),
-        //           onPress: () => BackgroundGeolocation.showAppSettings(),
-        //         },
-        //         {
-        //           text: languages.t('label.no'),
-        //           onPress: () => console.log('No Pressed'),
-        //           style: 'cancel',
-        //         },
-        //       ],
-        //     ),
-        //   1000,
-        // );
+        setTimeout(
+          () =>
+            Alert.alert(
+              languages.t('label.require_location_information_title'),
+              languages.t('label.require_location_information_message'),
+              [
+                {
+                  text: languages.t('label.yes'),
+                  onPress: () => BackgroundGeolocation.showAppSettings(),
+                },
+                {
+                  text: languages.t('label.no'),
+                  onPress: () => {
+                    console.log('No Pressed'), LocationServices.stop();
+                    //                     Settings.setIsLogging(false);
+                  },
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
       }
     });
   }
 
+  static isheadlessAlreadyRunning() {
```

Comment:
```suggestion
  static isHeadlessTaskRunning() {
```

Also - please add a small comment to the method.

Reply:
will do